src/golioth_openthread.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(golioth_openthread);
+
+#include <openthread/nat64.h>
+#include <openthread/dns_client.h>
+#include <zephyr/net/openthread.h>
+#include <openthread/error.h>
+
+static otDnsQueryConfig dns_query;
+static char *golioth_ipv6_addr;
+
+K_SEM_DEFINE(ot_dns_resolve, 0, 1);
```

Comment:
It is better to not use any global variables like that.

Please create a helper structure that you will pass using `void *aContext` to `ot_dns_callback()`:
```
struct ot_dns_resolve_context {
    struct k_sem sem;
    char *ipv6;
    otDnsQueryConfig query_config; /* optional, if you need it in the callback */
};
```
then access those members in `ot_dns_callback`.